File: src/main/java/com/wafflestudio/draft/dto/request/SignUpRequest.java
package com.wafflestudio.draft.dto.request;

import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.validation.constraints.NotNull;

@EqualsAndHashCode(callSuper = true)
@Data
public class SignUpRequest extends AuthenticationRequest {
    @NotNull
    private String username;
    @NotNull
    private String grantType;
    
    private String authProvider;
    private String accessToken;
    private String email;
    private String password;

    // TODO: More information about region, preference... should be added
}

File: src/main/kotlin/com/wafflestudio/draft/dto/request/SignUpRequest.kt
package com.wafflestudio.draft.dto.request

import com.wafflestudio.draft.dto.AuthenticationDTO

data class SignUpRequest(
        val username: String,
        override var grantType: String? = null,
        override var authProvider: String? = null,
        override var accessToken: String? = null,
        override var email: String? = null,
        override var password: String? = null // TODO: More information about region, preference... should be added
) : AuthenticationDTO.Request()
